ros/lib/utils.py

```diff
@@ -7,3 +7,21 @@ def is_valid_uuid(val):
         return True
     except ValueError:
         return False
+
+
+def get_or_create(session, model, keys, **kwargs):
+    if not keys:
+        keys = kwargs.keys()
+    if isinstance(keys, str):
+        keys = [keys]
+    if not isinstance(keys, list):
+        raise TypeError('keys argument must be a list or string')
+    instance = session.query(model).filter_by({k: kwargs[k] for k in keys}).first()
```

Comment:
Getting below error on this line. 
`ERROR  - run - An error occurred during message processing: TypeError('filter_by() takes 1 positional argument but 2 were given')`